### PR TITLE
templates: add k8s-setup-delete-old-node.service

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/11235

### CODE REVIEW

File: templates/master.yaml.tmpl [hunk 1]
```diff
@@ -1014,14 +1014,33 @@ systemd:
...
       [Install]
       WantedBy=multi-user.target
+  - name: k8s-setup-delete-old-node.service
+    enabled: true
+    contents: |
+      [Unit]
+      Description=Deletes old kubernetes node which this instance is going to replace
+      After=k8s-setup-download-hyperkube-binaries.service{{ if eq .Provider "aws" }} set-hostname.service{{ end }}
+      Requires=k8s-setup-download-hyperkube-binaries.service{{ if eq .Provider "aws" }} set-hostname.service{{ end }}
+      ConditionFirstBoot=yes
```
Comment (kopiczko) [inline templates/master.yaml.tmpl:1024]: Note it's fired only on first boot.

Comment (kopiczko) [reply templates/master.yaml.tmpl:1024]: I checked it isn't run on reboots and k8s-kubelet still starts.

**Comment by kopiczko** [comment]
Tested on ginger.

**Comment by calvix** [comment]
this doesn't work,  when  all master nodes are down or it's a fresh installation it cannot delete node since there is no API available and this causes k8s-kubelet not start since its dependency

  we need to rollback it and then try to run CI to ensure it works on fresh clusters

**Comment by kopiczko** [comment]
> this doesn't work, when all master nodes are down or it's a fresh installation it cannot delete node since there is no API available and this causes k8s-kubelet not start since its dependency
> 
> we need to rollback it and then try to run CI to ensure it works on fresh clusters

https://github.com/giantswarm/giantnetes-terraform/pull/339

**Comment by calvix** [comment]
thx

**Pull request merged**